File: suites/py_basic/greeter.py
# Copyright (c) 2026 The Blade Authors.
# All rights reserved.
#
# Author: chen3feng <chen3feng@gmail.com>

"""Trivial greeting helpers.

Kept intentionally minimal — this module exists so the py_basic suite
can exercise blade-build's py_library + py_test wiring end-to-end, not
to provide production logic.
"""

from __future__ import annotations


def greet(name: str) -> str:
    """Return ``"Hello, <name>!"`` for non-empty names, else ``"Hello, world!"``.

    Mirrors ``suites/cc_basic/hello.cc::Greet`` so the two suites stay
    symmetrical and easy to compare when diagnosing cross-language
    regressions.
    """
    target = name if name else "world"
    return f"Hello, {target}!"

File: suites/go_basic/greeter/greeter.go
// Copyright (c) 2026 The Blade Authors.
// All rights reserved.

// Package greeter provides a trivial greeting helper.
//
// Kept intentionally minimal — it exists so the go_basic suite can exercise
// blade-build's go_library / go_test / go_binary wiring end-to-end. Mirrors
// suites/py_basic/greeter.py and suites/cc_basic/hello.cc so the suites stay
// symmetrical when diagnosing cross-language regressions.
package greeter

// Greet returns "Hello, <name>!" for non-empty names, else "Hello, world!".
func Greet(name string) string {
	if name == "" {
		name = "world"
	}
	return "Hello, " + name + "!"
}
